core/src/main/java/org/elasticsearch/indices/recovery/StartRecoveryRequest.java

```diff
@@ -108,6 +114,7 @@ public void readFrom(StreamInput in) throws IOException {
         super.readFrom(in);
         recoveryId = in.readLong();
         shardId = ShardId.readShardId(in);
+        targetAllocationId = in.readString();
```

Comment:
yeah, i just removed this in a follow-up commit for master (ebad5a8). The first commit will be the one that goes into 5.4 and 5.x